src/main/java/com/wiley/autotest/selenium/elements/upgrade/BaseTeasyElement.java

```diff
@@ -0,0 +1,495 @@
+            builder.append("/..");
+        }
+        return TeasyElementWrapper.wrap(this, find(By.xpath(builder.toString())), By.xpath(builder.toString()), TeasyElementType.DOM);
+    }
+
+    @Override
+    public TeasyElement element(By by) {
+        return contextFinder.visibleElement(by);
+    }
+
+    @Override
+    public TeasyElement element(By by, SearchStrategy strategy) {
+        return customContextFinder(strategy).visibleElement(by);
+    }
+
+    @Override
+    public List<TeasyElement> elements(By by) {
+        return contextFinder.visibleElements(by);
+    }
+
+    @Override
+    public List<TeasyElement> elements(By by, SearchStrategy strategy) {
+        return customContextFinder(strategy).visibleElements(by);
+    }
+
+    @Override
+    public TeasyElement domElement(By by) {
+        return contextFinder.presentInDomElement(by);
+    }
+
+    @Override
+    public TeasyElement domElement(By by, SearchStrategy strategy) {
+        return customContextFinder(strategy).presentInDomElement(by);
+    }
+
+    @Override
+    public List<TeasyElement> domElements(By by) {
+        return contextFinder.presentInDomElements(by);
+    }
+
+    @Override
+    public List<TeasyElement> domElements(By by, SearchStrategy strategy) {
+        return customContextFinder(strategy).presentInDomElements(by);
+    }
+
+    TeasyElementFinder customContextFinder(SearchStrategy strategy) {
+        return new TeasyElementFinder(getWebDriver(), strategy, this);
+    }
+
+    @Override
+    public void click() {
+        try {
+            try {
+                wrappedElement.click();
+            } catch (StaleElementReferenceException e) {
+                clickForStaleElement();
+            } catch (UnhandledAlertException ignored) {
+                new Report("*****UnhandledAlertException***** during click! Doing nothing just trying to continue the test. Locator=" + locator
+                        .getLocator()).jenkins();
+            } catch (UnreachableBrowserException ignored) {
+                //doing this because for FireFox for some reason browser becomes unresponsive after click
+                //but actually it is alive so it worth to try to continue test
+                //it will fail on the next method after click if some real error happened
+                new Report("*****ERROR*****UnreachableBrowserException***** during click! Doing nothing just trying to continue the test. Locator=" + locator
+                        .getLocator()).jenkins();
+            } catch (ElementNotVisibleException needToScroll) {
+                clickForNeedToScroll();
+            } catch (WebDriverException ignoredOrNeedToScroll) {
+                clickForIgnoredScroll(ignoredOrNeedToScroll);
+            }
+        } catch (Exception e) {
+            new Report("*****UNKNOWN ERROR*****Exception***** DURING CLICK LOGIC. SHOULD BE REFACTORED!!!! Locator=" + locator.getLocator(), e).jenkins();
+        }
```

Comment:
Looks like this catch is unreachable. So we can avoid external try block

Reply:
The idea of it is to catch any exception that could be produced by any line inside inner try-catch. Although I agree it looks very unclear... but I would prefer not removing it, as it may lead to some other issues. Let's discuss on how we can rework it